main.go

```diff
@@ -43,9 +49,23 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
 
+	// Determine the namespace we're running in. Normally injected into the pod as an env
+	// var via the Kube downward API configured in the Deployment. We also default to "argocd"
+	// for developers running the binary locally who may not remember to set a NAMESPACE environment
+	// variable.
+	ns := os.Getenv("NAMESPACE")
+	if len(ns) == 0 {
+		ns = defaultNamespace
+	}
+	setupLog.Info("using argocd namespace", "namespace", ns)
```

Comment:
Maybe application set is better

Reply:
If we're destined to be incorporated in argocd it might be best to use the namespace where we expect to land. Although I'm not sure if argocd assumes anything about the namespace it's running in...

Reply:
Because we're unsure, why don't we remove the concept of a default namespace. If there is no NAMESPACE env var, we error out and ask you to set it, which would only affect developers running the binary locally on their systems. The pod definition will always ensure it's set to whatever the current namespace is.

Reply:
Pushed as a separate commit for your consideration.